src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -28,13 +28,32 @@ namespace OpenTelemetry.Metrics
     public class MeterProviderSdk
         : MeterProvider
     {
-        private ConcurrentDictionary<Meter, int> meters;
-        private MeterListener listener;
-        private CancellationTokenSource cts;
-        private Task observerTask;
+        private readonly CancellationTokenSource cts;
+        private readonly Task observerTask;
+        private readonly Task exporterTask;
+        private readonly ConcurrentDictionary<Meter, int> meters;
+        private readonly MeterListener listener;
 
-        internal MeterProviderSdk(IEnumerable<string> meterSources, int observationPeriodMilliseconds)
+        internal MeterProviderSdk(
+            IEnumerable<string> meterSources,
+            int observationPeriodMilliseconds,
+            int exportPeriodMilliseconds,
+            MeasurementProcessor[] measurementProcessors,
+            ExportMetricProcessor[] exportMetricProcessors)
         {
+            this.ObservationPeriodMilliseconds = observationPeriodMilliseconds;
+            this.ExportPeriodMilliseconds = exportPeriodMilliseconds;
+
+            // Setup our Processors
+
+            this.MeasurementProcessors.AddRange(measurementProcessors);
```

Comment:
To begin with, why don't we just have a single thing called Processor which gets notified for every instrument.record.
AggregatingProcessor, ExportingProcessor etc need not be specially handled by SDK for now.